src/swupd/manifest.go

```diff
@@ -0,0 +1,118 @@
+package swupd
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+)
+
+// ManifestHeader contains metadata for the manifest
+type ManifestHeader struct {
+	Format      int
+	Version     int32
+	Previous    int32
+	FileCount   int32
+	TimeStamp   time.Time
+	ContentSize int64
+	Includes    []Manifest
```

Comment:
These should be pointers to the Manifest structures. Otherwise we'll have to copy definitions around, imagine parsing os-core-update, that includes os-core previously parsed.

I would actually just keep the names of the manifests being included and let a later step that needs resolve into the actual references (i.e. Includes would be []string or a type based on string).

Reply:
Since at this point the only information we have about the includes is the name, I agree that this makes more sense as a []string. @icarus-sparry you were against this initially, did you think this should be a *[]Manifest?

Reply:
@cmarcelo is correct, we do want pointers to Manifests for the reasons he mentions.

We probably want a new type here, a structure with a name and a pointer to a manifest, the latter being filled in in the "resolve" step mentioned.